cfgov/v1/forms.py

```diff
@@ -1,16 +1,49 @@
 from django import forms
 from django.forms.widgets import DateInput, SelectMultiple
+from sheerlike.templates import date_formatter
 from v1.models.events import EventPage
+from django.forms.utils import ErrorList
 
+class FilterErrorList(ErrorList):
+
+    def __str__(self):
+        return '\n'.join(str(e) for e in self)
+
+class FormDateField(forms.DateField):
+    def to_python(self, value):
+        if value:
+            try:
+                value = date_formatter(value)
+            except Exception as e:
+                raise forms.ValidationError(('"%s" is not a valid Date'), params=(value,))
+
+
+        return super(FormDateField, self).to_python(value)
 
 class CalenderPDFFilterForm(forms.Form):
-    filter_calendar = forms.CharField()
-    filter_range_date_gte = forms.DateField(input_formats=['%Y-%m-%d'])
-    filter_range_date_lte = forms.DateField(input_formats=['%Y-%m-%d'])
+    filter_calendar = forms.CharField(label='Calendars')
+    filter_range_date_gte = FormDateField(label='From', required=False)
+    filter_range_date_lte = FormDateField(label='To',required=False)
+
+    def __init__(self, *args, **kwargs):
+        kwargs['error_class'] = FilterErrorList
+        super(CalenderPDFFilterForm, self).__init__(*args, **kwargs)
 
     def clean_filter_calendar(self):
         return self.cleaned_data['filter_calendar'].replace(' ', '+')
 
+    def clean(self):
+        cleaned_data = super(CalenderPDFFilterForm, self).clean()
+        from_date = cleaned_data['filter_range_date_gte'] \
+                    if 'filter_range_date_gte' in cleaned_data else False
+        to_date = cleaned_data['filter_range_date_lte'] \
+                  if 'filter_range_date_lte' in cleaned_data else False
+
+        if from_date == None and to_date == None :
```

Comment:
Fixed in https://github.com/cfpb/cfgov-refresh/commit/606b7c1d369eb210b18ded6cb7f660ffbe3bf87e